### PR TITLE
[SG-682] Add Event Log Entries to Organization Domain

### PR DESCRIPTION
**Pull request by gbubemismith**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

The essence of this PR is to record events that take place during domain claiming by an admin, From creating a domain entry for an organization, removing a domain entry and verifying a domain. This PR also includes changes to the implementation of how a domain entry is created, and a new column, `LastDateChecked` is added to the `OrganizationDomain` table to track the exact time the background service was run. 

## Code changes

**Commit messages can be used to review this PR.**

* **src/Core/Entities/Event.cs:** The `DomainName` property is added to the object to capture the domain name when an event is logged.
* **src/Api/Models/Response/EventResponseModel.cs:** The `DomainName` property is added to the response model.
* **src/Core/Enums/EventType.cs** Added the organization domain event types to the enum.
* **src/Core/Enums/EventSystemUser.cs:** Added a new system user enum `SSO` to capture verification attempts done by the system.
* **src/Core/Services/Implementations/EventService.cs:** Implementation of methods used in capturing `OrganizationDomain` specific events.
* **src/Sql/dbo/Stored Procedures/Event_Create.sql:** Stored procedure updated with the new `DomainName` property
* **src/Sql/dbo/Tables/Event.sql:** Event table updated with a new column
* **util/Migrator/DbScripts/2022-12-08_00_EventsDomainName.sql:** mssql migratory script to alter events table and add a new column
* **src/Core/OrganizationFeatures/OrganizationDomains/CreateOrganizationDomainCommand.cs:** Command updated with log events for adding a new domain for an organization and logging the verification status for the domain verification.
* **src/Core/OrganizationFeatures/OrganizationDomains/DeleteOrganizationDomainCommand.cs:** Logged removed domain event
* **src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:** Logged verification domain event

**Changes not related to events log**
* **src/Core/OrganizationFeatures/OrganizationDomains/CreateOrganizationDomainCommand.cs:** Added check to ensure an organization cannot create duplicate domain entries.
* **src/Core/Exceptions/DuplicateDomainException.cs** Duplicate domain entry exception
* **src/Core/Entities/OrganizationDomain.cs:** Added `lastCheckedDate` property to tell the last time a verification attempt was made.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: util/MySqlMigrations/Scripts/2022-12-09_00_OrganizationDomainLastCheckedDate.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿START TRANSACTION;
+
+ALTER TABLE `OrganizationDomain` ADD `LastCheckedDate` datetime(6) NULL;
```
Comment (JaredSnider-Bitwarden) [inline util/MySqlMigrations/Scripts/2022-12-09_00_OrganizationDomainLastCheckedDate.sql:3]: So, above, in the `util/Migrator/DbScripts/2022-11-03_00_OrganizationDomainInit.sql` file, the length of the datetime was 7, and here it is 6?  Is this correct b/c MySql is 0 based and the other DB was not? 

image

**Review by JaredSnider-Bitwarden — COMMENTED** [review]
It looked great overall. However, I think a review by someone with a bit more familiarity with the events DB / architecture would be a good idea.

File: util/MySqlMigrations/Scripts/2022-12-09_00_OrganizationDomainLastCheckedDate.sql (hunk 1 above)
Comment (gbubemismith) [reply util/MySqlMigrations/Scripts/2022-12-09_00_OrganizationDomainLastCheckedDate.sql:3]: Yeah that's the MySQL equivalent

**Review by differsthecat — APPROVED** [review]
Looks great!
@JaredSnider-Bitwarden had some good feedback that you took care of!

**Review by JaredSnider-Bitwarden — APPROVED** [review]
👍

**Pull request merged**